feature/swt-78 add time

jennyknuth opened the pull request:
This changes the standard date input into a date-time picker with validation. 

demo: `npm install && npm start`

jennyknuth committed:
rough in dateTime picker

jennyknuth committed:
get dateTime picker working and style

jennyknuth committed:
remove console errors and warnings and update tests

greglucas commented on src/app/containers/visualizer/visualizer.container.ts:
```diff
@@ -289,7 +280,7 @@ export class VisualizerComponent implements OnInit {
     }
 
-    setF107Values( data: number[], date: moment.Moment ) {
+    setF107Values( data: number[], timestamp: number ) {
```
I'm not following why you changed this over to a number now, rather than keeping it as a Moment. It seems like you're calling `moment.utc()` in multiple places now. It seems like it would be better to store the moment and then do the string/number cast once initially if possible?

greglucas reviewed (approved):
Very nice! Just some minor comments/questions below.

* When changing the date, the UTC time gets forced to "00:00:00", is this expected? I guess my preference would be to keep that left alone if possible. If it would add a lot of complication, it probably isn't worth it though.

* Right now I can't submit "HH:mm", it requires the full seconds as well. Is there a way to get rid of the seconds? That seems like too fine of detail here.

jennyknuth commented:
Both (all?) issues have to do with using our dateTime library component. I changed from a plain input to our dateTime component that includes validation, etc. The component comes with HH:mm:ss (not a customizable parameter right now). It outputs timestamps instead of moment dates (internally it uses moment, but the output right now is numbers). Before I converted things to numbers in a few places, now I convert them to moment objects in a few places. I don't think one way was clearly better than the other—in general numbers are easier to deal with. And yes, it does reset to the beginning of the day when you select a date, it does not take note of the previous time and then add that back. All of these are good suggestions for future improvements on that component.

greglucas commented:
Thanks, @jennyknuth! I figured it was a limitation of the dependencies, so you're good to merge this and we can keep those updates in mind for adding to LASP's datetime picker upstream.

My preference for keeping moment instead of number: I like keeping the most descriptive object possible, because moment indicates there is extra information attached (like timezone, that it is a date, etc...), whereas with Number you've lost information and have to keep track of what exactly a Number is (relative to some epoch? Is it in seconds, minutes? etc...)
